### PR TITLE
ci: fix docker compose call

### PR DESCRIPTION
**Pull request by fzipi**
Fixes docker compose call.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,11 +7,6 @@ on:
       - 'rules/**'
       - 'tests/**'
       - '.github/**'
-  pull_request:
```
Comment (theseion) [inline .github/workflows/test.yml:10]: Why did you remove this?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -33,9 +28,10 @@ jobs:
       - name: "Run tests for ${{ matrix.modsec_version }}"
         run: |
           mkdir -p tests/logs/${{ matrix.modsec_version }}/{nginx,apache2}
-          docker-compose -f ./tests/docker-compose.yml up -d "${{ matrix.modsec_version }}"
-          docker-compose -f ./tests/docker-compose.yml logs
+          docker compose -f ./tests/docker-compose.yml up -d "${{ matrix.modsec_version }}"
+          docker compose -f ./tests/docker-compose.yml logs
           [ $(docker inspect ${{ matrix.modsec_version }} --format='{{.State.Running}}') = 'true' ]
+          chmod o+w -R ./tests/logs
```
Comment (theseion) [inline .github/workflows/test.yml:34]: Should not be necessary, as we use the root user in the compose file.

**Review by theseion — DISMISSED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (fzipi) [reply .github/workflows/test.yml]: Just trying to get it working. It worked once after doing chmod, but on push. On pull_request it failed to run the pipeline. 🤷

File: tests/docker-compose.yml [hunk 3]
```diff
@@ -1,36 +1,30 @@
...
       BACKEND: http://backend
-      PORT: "80"
-      MODSEC_RULE_ENGINE: DetectionOnly
-      PARANOIA: 4
-      TZ: "${TZ}"
-      ERRORLOG: "/var/log/error.log"
-      ACCESSLOG: "/var/log/access.log"
+      COMBINED_FILE_SIZES: "65535"
+      CRS_ENABLE_TEST_MARKER: 1
+      ERRORLOG: "/var/log/apache2/error.log"
```
Comment (theseion) [inline tests/docker-compose.yml:10]: The workflow creates `/var/log/modsec2-apache/apache2`.

Comment (fzipi) [reply tests/docker-compose.yml:10]: Hmmm... it should not, volumes say:
```
     volumes:
       - ./logs/modsec2-apache:/var/log/apache2:rw
```

Comment (fzipi) [reply tests/docker-compose.yml:10]: Wait. Got it.

**Comment by fzipi** [comment]
WTH, this works locally.

**Comment by fzipi** [comment]
Ok, finally. It was permissions in the end. The `user: root` wasn't there... as expected.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -31,29 +31,29 @@ jobs:
...
         if: failure()
         with:
           name: waf-logs
           path: tests/logs/${{ matrix.modsec_version }}
 
       - name: Clean docker-compose
         run: |
-          docker-compose -f ./tests/docker-compose.yml stop "${{ matrix.modsec_version }}"
-          docker-compose -f ./tests/docker-compose.yml down
+          docker compose -f ./tests/docker-compose.yml stop "${{ matrix.modsec_version }}"
```
Comment (theseion) [inline .github/workflows/test.yml:58]:
```suggestion
          docker compose -f ./tests/docker-compose.yml down -t 0
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -31,29 +31,29 @@ jobs:
...
+          docker compose -f ./tests/docker-compose.yml down
```
Comment (theseion) [inline .github/workflows/test.yml:59]:
```suggestion
```

File: tests/docker-compose.yml [hunk 6]
```diff
@@ -1,67 +1,70 @@
...
-
 services:
   modsec2-apache:
     container_name: modsec2-apache
     image: owasp/modsecurity-crs:apache
+    # NOTE: The user used to run the container process is explicitly set to
+    # 'root'. This fixes issues with permissions on the logging directories used
+    # as bind mounts. This is done as *a convenience for running the CRS testing
+    # setup only* and *should not be done in general!*
+    user: root
```
Comment (theseion) [inline tests/docker-compose.yml:9]: Why is this showing as a new change? I had added this weeks ago...

**Review by theseion — CHANGES_REQUESTED** [review]

File: tests/docker-compose.yml (hunk 6 above)
Comment (fzipi) [reply tests/docker-compose.yml:9]: In the v3/dev branch?

Comment (theseion) [reply tests/docker-compose.yml:9]: Oh, I didn't see that.

**Review by theseion — APPROVED** [review]
